subsys/net/lib/nrf_cloud/include/nrf_cloud_codec.h

```diff
@@ -26,6 +26,10 @@ int nrf_cloud_encode_ua(const struct nrf_cloud_ua_param *param,
 int nrf_cloud_encode_sensor_data(const struct nrf_cloud_sensor_data *input,
 				 struct nrf_cloud_data *output);
 
+/**@brief Encode the sensor data to be sent to the device shadow. */
+int nrf_cloud_encode_shadow_data(const struct nrf_cloud_sensor_data *sensor,
```

Comment:
Could we use `struct nrf_cloud_data` here instead to make it agnostic to type of payload?

Reply:
I agree, it should be agnostic but it would require some changes to the API.

Right now we use the `type` member of the sensor struct as the string name for the JSON object. We could declare the object name through different methods, but I am leaning towards leaving it as is for this PR.